hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/TestContainerBalancerOperations.java

```diff
@@ -178,4 +180,44 @@ public void testIfCBCLIOverridesConfigs() throws Exception {
     running = containerBalancerClient.getContainerBalancerStatus();
     assertFalse(running);
   }
+
+  /**
+   * Tests that stopBalancer is idempotent - once the balancer is in STOPPED state,
+   * invoking stop again should be a no-op and return successfully with exit code 0.
+   */
+  @Test
+  public void testStopBalancerIdempotent() throws IOException {
+    boolean running = containerBalancerClient.getContainerBalancerStatus();
+    assertFalse(running);
+
+    Optional<Double> threshold = Optional.of(0.1);
+    Optional<Integer> iterations = Optional.of(10000);
+    Optional<Integer> maxDatanodesPercentageToInvolvePerIteration =
+        Optional.of(100);
+    Optional<Long> maxSizeToMovePerIterationInGB = Optional.of(1L);
+    Optional<Long> maxSizeEnteringTargetInGB = Optional.of(6L);
+    Optional<Long> maxSizeLeavingSourceInGB = Optional.of(6L);
+    Optional<Integer> balancingInterval = Optional.of(70);
+    Optional<Integer> moveTimeout = Optional.of(65);
+    Optional<Integer> moveReplicationTimeout = Optional.of(50);
+    Optional<Boolean> networkTopologyEnable = Optional.of(false);
+    Optional<String> includeNodes = Optional.of("");
+    Optional<String> excludeNodes = Optional.of("");
+    containerBalancerClient.startContainerBalancer(threshold, iterations,
+        maxDatanodesPercentageToInvolvePerIteration,
+        maxSizeToMovePerIterationInGB, maxSizeEnteringTargetInGB,
+        maxSizeLeavingSourceInGB, balancingInterval, moveTimeout,
+        moveReplicationTimeout, networkTopologyEnable, includeNodes,
+        excludeNodes);
+    running = containerBalancerClient.getContainerBalancerStatus();
+    assertTrue(running);
+
+    containerBalancerClient.stopContainerBalancer();
+    running = containerBalancerClient.getContainerBalancerStatus();
+    assertFalse(running);
+
+    // Calling stop balancer multiple times should not throw an exception
+    assertDoesNotThrow(() -> containerBalancerClient.stopContainerBalancer());
+    assertDoesNotThrow(() -> containerBalancerClient.stopContainerBalancer());
```

Comment:
Instead of calling stop twice here, let's call it once here and once after line 191 when balancer hasn't been started at all.